specs/experimental/gov-delegation.md

```diff
@@ -0,0 +1,440 @@
+# GovernanceDelegation
+
+<!-- START doctoc generated TOC please keep comment here to allow auto update -->
+<!-- DON'T EDIT THIS SECTION, INSTEAD RE-RUN doctoc TO UPDATE -->
+**Table of Contents**
+
+- [Overview](#overview)
+- [Interface](#interface)
+  - [Core Functions](#core-functions)
+    - [`delegate`](#delegate)
+    - [`delegateFromToken`](#delegatefromtoken)
+    - [`delegateBatched`](#delegatebatched)
+    - [`afterTokenTransfer`](#aftertokentransfer)
+    - [`migrateAccounts`](#migrateaccounts)
+    - [`_delegate`](#_delegate)
+  - [Getters](#getters)
+    - [`checkpoints`](#checkpoints)
+    - [`numCheckpoints`](#numcheckpoints)
+    - [`delegates`](#delegates)
+    - [`getVotes`](#getvotes)
+    - [`getPastVotes`](#getpastvotes)
+    - [`getPastTotalSupply`](#getpasttotalsupply)
+    - [`migrated`](#migrated)
+    - [`delegations`](#delegations)
+  - [Events](#events)
+    - [`DelegationCreated`](#delegationcreated)
+    - [`DelegateVotesChanged`](#delegatevoteschanged)
+- [Storage](#storage)
+- [Constants](#constants)
+- [Types](#types)
+  - [`Delegation`](#delegation)
+  - [`AllowanceType`](#allowancetype)
+  - [`DelegationAdjustment`](#delegationadjustment)
+  - [`Op`](#op)
+- [Backwards Compatibility](#backwards-compatibility)
+- [Migration](#migration)
+- [User Flow](#user-flow)
+  - [Partial Delegations](#partial-delegations)
+  - [Absolute & Relative Delegations](#absolute--relative-delegations)
+  - [Differences](#differences)
+- [Delegation Validation](#delegation-validation)
+- [Security Considerations](#security-considerations)
+  - [Dependence on GovernanceDelegation](#dependence-on-governancedelegation)
+  - [Connection with GovernanceToken](#connection-with-governancetoken)
+
+<!-- END doctoc generated TOC please keep comment here to allow auto update -->
+
+## Overview
+
+| Constant | Value                                        |
+|----------|----------------------------------------------|
+| Address  | `0x4200000000000000000000000000000000000043` |
+
+The `GovernanceDelegation` contract implements advanced delegation for the [`GovernanceToken`](gov-token.md).
+Advanced delegation allows for partial and relative delegations of voting power.
+
+The `GovernanceDelegation` contract migrates the delegation state from the `GovernanceToken` to itself
+through a hook-based approach. Specifically, the `GovernanceToken` calls the `GovernanceDelegation` contract's
+`afterTokenTransfer` function after a token transfer. This enables the `GovernanceDelegation` contract to consume
+the hook and update its delegation and checkpoint mappings accordingly. If either address involved in the transfer
+(`_from_` or `_to`) has not been migrated to the `GovernanceDelegation` contract, the contract copies the address'
+checkpoint data from the `GovernanceToken` to its own state.
+
+## Interface
+
+### Core Functions
+
+#### `delegate`
```

Comment:
Does this function overwrite the set of delegations or add to it?

Reply:
overwites, added clarification